### PR TITLE
[mock_tests] Add Sflow Orch UTs

### PR DESCRIPTION
**Pull request by ayurkiv-nvda**
Signed-off-by: Andriy Yurkiv <ayurkiv@nvidia.com>

**What I did**
Added Sflow gtest

**Why I did it**
Improve sflow orcagent coverage 

**How I verified it**
```
ayurkiv@487e531606e9:/sonic/src/sonic-swss/tests/mock_tests$ ./tests --gtest_filter=SflowOrchTest*
Running main() from /build/googletest-YnT0O3/googletest-1.10.0.20201025/googletest/src/gtest_main.cc
Note: Google Test filter = SflowOrchTest*
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from SflowOrchTest
[ RUN      ] SflowOrchTest.SflowEnableDisable
[       OK ] SflowOrchTest.SflowEnableDisable (46 ms)
[ RUN      ] SflowOrchTest.SflowCreateDelete
[       OK ] SflowOrchTest.SflowCreateDelete (46 ms)
[----------] 2 tests from SflowOrchTest (92 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test suite ran. (93 ms total)
[  PASSED  ] 2 tests.
```

**Details if related**

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@dgsudharsan could you please help to review ?

**Review by nazariig — APPROVED** [review]

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: tests/mock_tests/sfloworh_ut.cpp [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+                            {"admin_state", "up"},
+                        }
+                    }
+                });
+            mock_orch.doSflowTableTask(table4);
+            ASSERT_FALSE(mock_orch.get().m_sflowStatus);
+        }
+    }
+}
+
```
Comment (vivekreddynv) [inline tests/mock_tests/sfloworh_ut.cpp:371]: Can you add a few cases for notifications coming from APP_SFLOW_SESSION_TABLE_NAME & APP_SFLOW_SAMPLE_RATE_TABLE_NAME tables?

Here is a reference on testing if sai api's are called as expected: https://github.com/Azure/sonic-swss/blob/master/tests/mock_tests/portsorch_ut.cpp#L92

File: tests/mock_tests/sfloworh_ut.cpp [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+                    {
+                        "global",
+                        SET_COMMAND,
+                        {
+                            {"admin_state", "up"},
+                        }
+                    }
+                });
+            mock_orch.doSflowTableTask(table3);
+            ASSERT_TRUE(mock_orch.get().m_sflowStatus);
```
Comment (vivekreddynv) [inline tests/mock_tests/sfloworh_ut.cpp:353]: I think it's better to either one of `ASSERT_TRUE(Portal::SflowOrchInternal::getSflowStatusEnable(mock_orch.get()))` or  `ASSERT_TRUE(mock_orch.get().m_sflowStatus);` to keep the implementation consistent

**Review by vivekreddynv — CHANGES_REQUESTED** [review]

**Comment by liat-grozovik** [comment]
@vadymhlushko-mlnx coulld you please review review comments?

**Comment by volodymyrsamotiy** [comment]
> @vadymhlushko-mlnx coulld you please review review comments?

@ayurkiv-nvda will comments since it his PR

File: tests/mock_tests/sfloworh_ut.cpp [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+
+        void TearDown() override
+        {
+            this->deinitOrch();
+            this->deinitSwitch();
+            this->deinitSaiApi();
+        }
+
+    private:
+        void initSaiApi()
```
[the remark this answers is not in the document]
Comment (ayurkiv-nvda) [reply tests/mock_tests/sfloworh_ut.cpp:100]: you propose to create common parent class and use it for all tests in mock_tests directory? 
It sounds like a really good idea, but it looks like a separate task that is not in scope of these changes.

File: tests/mock_tests/sfloworh_ut.cpp (hunk 1 above)
Comment (vivekreddynv) [reply tests/mock_tests/sfloworh_ut.cpp:371]: Not sure about the scope though 
But i agree, the test can certainly be improved later on demand.

**Review by dgsudharsan — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@vivekreddynv can we move forward with the merge?

**Comment by vivekreddynv** [comment]
> @vivekreddynv can we move forward with the merge?

yes, i'll approve the PR

**Review by vivekreddynv — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**